### PR TITLE
Implement batch transactions RPC endpoint.

### LINKED ISSUE #1227: Implement batch transactions RPC endoint
As part of EIP-1559 testnet work, we are facing some issues regarding transaction orchestration.
We want to simulate 25M blocks for long periods.
A dedicated endpoint to submit batch of transactions would help achieving this goal.

### PR DESCRIPTION
**Pull request by abdelhamidbakhta**
## PR description
As part of EIP-1559 testnet work, we are facing some issues regarding transaction orchestration.
We want to simulate 25M blocks for long periods.
A dedicated endpoint to submit batch of transactions would help achieving this goal.
This PR introduces a new RPC endpoint `debug_batchSendRawTransaction`.
### Request parameters
`DATA`, The signed transaction data array.

`params: ["0xf868808203e882520894627306090abab3a6e1400e9345bc60c78a8bef57872386f26fc10000801ba0ac74ecfa0e9b85785f042c143ead4780931234cc9a032fce99fab1f45e0d90faa02fd17e8eb433d4ca47727653232045d4f81322619c0852d3fe8ddcfcedb66a43", "0xf868018203e882520894627306090abab3a6e1400e9345bc60c78a8bef57872386f26fc10000801ca0b24ea1bee8fe36984c36acbf80979a4509f23fc17141851e08d505c0df158aa0a00472a05903d4cd7a811bd4d5c59cc105d93f5943f3393f253e92e65fc36e7ce0"]`

### Response
Returns the status of the execution for each input raw transaction. Individual result object contains:

- `index`: the index of the transaction in the request parameters array,
- `success`: a boolean indicating whether or not the transaction has been added to the transaction pool, and
- `errorMessage`: an optional error message.

## Fixed Issue(s)
Fixes #1227 
## Changelog
- Added `debug_batchSendRawTransaction` JSON-RPC API to submit multiple signed transactions with a single call.

#### Sample request

```json
{
   "jsonrpc":"2.0",
   "method":"debug_batchSendRawTransaction",
   "params":[
      "0xf868808203e882520894627306090abab3a6e1400e9345bc60c78a8bef57872386f26fc10000801ba0ac74ecfa0e9b85785f042c143ead4780931234cc9a032fce99fab1f45e0d90faa02fd17e8eb433d4ca47727653232045d4f81322619c0852d3fe8ddcfcedb66a43",
      "0x416",
      "0xf868018203e882520894627306090abab3a6e1400e9345bc60c78a8bef57872386f26fc10000801ca0b24ea1bee8fe36984c36acbf80979a4509f23fc17141851e08d505c0df158aa0a00472a05903d4cd7a811bd4d5c59cc105d93f5943f3393f253e92e65fc36e7ce0",
      "0xf868808203e882520894627306090abab3a6e1400e9345bc60c78a8bef5787470de4df820000801ca0f7936b4de04792e3c65095cfbfd1399d231368f5f05f877588c0c8509f6c98c9a01834004dead527c8da1396eede42e1c60e41f38a77c2fd13a6e495479c729b99"
   ],
   "id":1
}
```

#### Sample response

```json
{
  "jsonrpc": "2.0",
  "id": 1,
  "result": [
    {
      "index": 0,
      "success": true
    },
    {
      "index": 1,
      "success": false,
      "errorMessage": "Invalid raw transaction hex"
    },
    {
      "index": 2,
      "success": true
    },
    {
      "index": 3,
      "success": false,
      "errorMessage": "TRANSACTION_REPLACEMENT_UNDERPRICED"
    }
  ]
}
```

Signed-off-by: Abdelhamid Bakhta <abdelhamid.bakhta@consensys.net>

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/DebugBatchSendRawTransaction.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+  @Override
+  public String getName() {
+    return RpcMethod.DEBUG_BATCH_RAW_TRANSACTION.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
+    final List<ExecutionStatus> executionStatuses = new ArrayList<>();
+    IntStream.range(0, requestContext.getRequest().getParamLength())
```
Comment (shemnon) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/DebugBatchSendRawTransaction.java:57]: So if all of the TXes fail the RPC call succeeds, but with an array of each failed TX?  That's fine, no need to throw a JSON-RPC error in that case.

Comment (abdelhamidbakhta) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/DebugBatchSendRawTransaction.java:57]: Yes, the RPC call would succeed with an array of each failed TX.

**Comment by abdelhamidbakhta** [comment]
> Needs some test cases. I would be satisfied with JSON only tests in `ethereum/api/src/test/resources/org/hyperledger/besu/ethereum/api/jsonrpc/eth`, principally so we can see the input and output json. But feel free to write mock based tests if you think they would add value.

Yes, i will add some test cases. I wanted to have a review for the overall idea first.

**Comment by shemnon** [comment]
Ok.  In general it looks reasonable.  I expect that if there are any issues they would be the corner cases that tests should expose.

**Comment by abdelhamidbakhta** [comment]
> Maybe a case with a bad TX signature? It should be as simple as changing the right hex digit on a valid tx.

Done.

**Pull request merged**